src/Illuminate/Auth/Notifications/VerifyEmail.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Illuminate\Auth\Notifications;
+
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Facades\URL;
+use Illuminate\Support\Facades\Lang;
+use Illuminate\Notifications\Notification;
+use Illuminate\Notifications\Messages\MailMessage;
+
+class VerifyEmail extends Notification
+{
+    /**
+     * The callback that should be used to build the mail message.
+     *
+     * @var \Closure|null
+     */
+    public static $toMailCallback;
+
+    /**
+     * Get the notification's channels.
+     *
+     * @param  mixed  $notifiable
+     * @return array|string
+     */
+    public function via($notifiable)
+    {
+        return ['mail'];
+    }
+
+    /**
+     * Build the mail representation of the notification.
+     *
+     * @param  mixed  $notifiable
+     * @return \Illuminate\Notifications\Messages\MailMessage
+     */
+    public function toMail($notifiable)
+    {
+        if (static::$toMailCallback) {
+            return call_user_func(static::$toMailCallback, $notifiable);
+        }
+
+        return (new MailMessage)
+            ->subject(Lang::getFromJson('Verify Email Address'))
+            ->line(Lang::getFromJson('Please click the button below to verify your email address.'))
+            ->action(
+                Lang::getFromJson('Verify Email Address'),
+                $this->verificationUrl($notifiable)
+            )
+            ->line(Lang::getFromJson('If you did not create an account, no further action is required.'));
+    }
+
+    /**
+     * Get the verification URL for the given notifiable.
+     *
+     * @param  mixed  $notifiable
+     * @return string
+     */
+    protected function verificationUrl($notifiable)
+    {
+        return URL::temporarySignedRoute(
+            'verification.verify', Carbon::now()->addMinutes(60), ['id' => $notifiable->getKey()]
```

Comment:
Does this mean the verification link is valid only for 60 minutes? If it expires, how would one obtain a new one? What is the benefit of making this route time-dependent?

Reply:
It will help to prevent verification hash bruteforcing.

Reply:
Can this be a configurable value?

Reply:
@a-komarev I see... Thanks!